src/pages/RecordPage.vue

```diff
@@ -1,84 +1,109 @@
 <script>
-import { mapGetters, mapActions } from 'vuex';
+import { ref, computed, onMounted, onUnmounted } from 'vue';
 import { useVuelidate } from '@vuelidate/core';
+import { useStore } from 'vuex';
 import { required, minValue } from '@vuelidate/validators';
+import { useCategory } from '@/hooks/useCategory';
+import { useRecord } from '@/hooks/useRecord';
+import { useModal } from '@/hooks/useModal';
 
 export default {
   setup() {
-    return { v$: useVuelidate() };
-  },
-  data: () => ({
-    categories: [],
-    loading: true,
-    select: null,
-    currentCategoryId: null,
-    type: 'outcome',
-    amount: null,
-    description: null,
-  }),
-  computed: {
-    ...mapGetters(['currentUserInfo']),
-  },
-  methods: {
-    ...mapActions(['fetchCategories', 'addNewRecord', 'updateUserInfo']),
-    async onSubmitRecordCreate() {
-      if (this.v$.$invalid) {
-        this.v$.$touch();
+    const { showErrorModal } = useModal();
+    const { fetchCategories } = useCategory();
+    const { addNewRecord } = useRecord();
+    const store = useStore();
+
+    const currentUserInfo = computed(() => store.getters.currentUserInfo);
+
+    const categorySelect = ref(null);
+    const categories = ref([]);
+    const loading = ref(true);
+    const select = ref(null);
+    const currentCategoryId = ref(null);
+    const type = ref('outcome');
+    const amount = ref(null);
+    const description = ref(null);
+
+    const amountMin = ref(1);
+    const rules = computed(() => ({
+      amount: {
+        required,
+        minValue: minValue(amountMin.value),
+      },
+      description: {
+        required,
+      },
+    }));
+    const v$ = useVuelidate(rules, { amount, description });
+
+    const onSubmitRecordCreate = async () => {
+      if (v$.value.$invalid) {
+        v$.value.$touch();
         return;
       }
 
-      if (this.type === 'outcome' && this.amount > this.currentUserInfo.bill) {
-        this.$message(`На счете не хватает: ${this.currentUserInfo.bill - this.amount}`);
+      if (type.value === 'outcome' && amount.value > currentUserInfo.value.bill) {
```

Comment:
```suggestion
      const isOutcome = type.value === 'outcome'
      const isInsufficientFunds = amount.value > currentUserInfo.value.bill
      const isError = isOutcome && isInsufficientFunds
       if (isError) {
```
Немного легче прочитать